api/app/tests/hfi/test_hfi_endpoint_request.py

```diff
@@ -44,9 +44,53 @@ def mock_get_fire_weather_stations(_):
     mock_station_crud(monkeypatch)
 
 
+def _setup_mock_with_role(monkeypatch: pytest.MonkeyPatch, role: str):
+    """ Prepare jwt decode to be mocked with permission
+    """
+    _setup_mock(monkeypatch)
+
+    class MockJWTDecodeWithRole:
+        """ Mock pyjwt module with role """
+
+        def __init__(self, role):
+            self.decoded_token = {
+                "preferred_username": "test_username",
+                "resource_access": {
+                    "wps-web": {
+                        "roles": [
+                            role
+                        ]
+                    }
+                }}
+
+        def __getitem__(self, key):
+            return self.decoded_token[key]
+
+        def get(self, key, _):
+            "Returns the mock decoded token"
+            return self.decoded_token[key]
+
+        def decode(self):
+            "Returns the mock decoded token"
+            return self.decoded_token
+
+    def mock_fire_start_role_function(*args, **kwargs):  # pylint: disable=unused-argument
+        return MockJWTDecodeWithRole(role)
+
+    if(role != 'None'):
+        monkeypatch.setattr("jwt.decode", mock_fire_start_role_function)
+
+
+@pytest.mark.usefixtures('mock_jwt_decode')
+@scenario('test_hfi_endpoint_request.feature', 'HFI - GET request')
+def test_fire_behaviour_calculator_get_scenario():
+    """ BDD Scenario. """
+    pass
+
+
 @pytest.mark.usefixtures('mock_jwt_decode')
```

Comment:
Can't can drop @pytest.mark.usefixtures('mock_jwt_decode') since we're patching it in the scenario?

Reply:
I still need it here for the station select tests that have no permissions yet